pkg/monitors/postgresql/monitor.go

```diff
@@ -93,10 +98,12 @@ func (m *Monitor) Configure(conf *Config) error {
 
 	queriesGroupEnabled := m.Output.HasEnabledMetricInGroup(groupQueries)
 
-	connStr, err := conf.connStr()
+	connStr, host, port, err := conf.connStr()
 	if err != nil {
 		return fmt.Errorf("could not render connectionString template: %v", err)
 	}
+	m.Output.AddExtraDimension("postgres_host", host)
```

Comment:
Yeah I would get rid of these (host/port).  If you are using the docker or k8s observer, the observer will put dimensions on the output that will identify it based on certain attributes of the postgresql container.  If you are using a plain host without containers, the `host` dimension already added to datapoints at a higher level can be used to differentiate postgresql servers.  If you are targeting a postgresql server from an agent on a separate host, this could be helpful in combination with the `disableHostDimensions: true` config option on the monitor, but the standard deployment model is to run the agent on the same host.  So basically these dimensions will be redundant or useless in a lot of scenarios and shouldn't be added universally.

Reply:
I understand and this is true that we do not need in most cases.

However this will allow to support every cases natively. Adding extra dimensions is not a problem in my opinion but this need is not obvious for everybody so it will avoid mishap for some users and ease the config for everyone.

I would say, this change should be fully retro compatible as it does not delete any dimension or change existing ones (so detectors using explicit aggregation will not be impacted) and just add more granularity (at worst redundant) so this will also work when aggregation is not used.

Also, some existing monitors work on this model (i.e. collectd/redis, elasticsearch with `plugin_instance` dimension). Some also provide an explicit option to set this field like apache or nginx with `name` which is more natural than extraDimension common monitor parameter.

I understand for most of them this is a collectd legacy but this make it straightforward for them to be used and this is sad to "lose" this feature in new "signalfx" monitors versions whereas I very like the default behavior of signalfx to apply signalflow to every single timeseries found contrary to datadog where aggregation is mandatory.

In my opinion, even if redondant in general it only brings good things so I would prefer to keep it if you do not have strong opinion on this. Else, tell me again and I will remove this change.

Reply:
I think the best thing is if you just always made the `host` dimension refer to the host of the postgresql server and refer to that in your charts/detectors.  If you are using the standard deployment model where the agent runs on the same host as Postgres, then it will already be the case.  If you are monitoring a remote server like RDS, then the best thing to do is to set `disableHostDimensions: true` on the monitor and then set `extraDimensions: {host: *host}` in your config.  Otherwise you'll get `host` dimension set to the host of the agent, which is probably irrelevant to monitoring purposes.  I admit it is not the most natural thing to have to configure it that way but it is because it is going against the standard deployment model of the agent, which always gets favored for consistency.

The other benefit of identifying postgres servers by `host` is that you can correlate other metrics from other monitors by that same dimension.  Nothing else is going to have `postgres_host` on it.

Reply:
ok it is removed!